design.md

```diff
@@ -20,13 +21,21 @@
   - Users can choose whether the kernel should connect to real hardware or to a simulator.
     - If hardware is selected, the user must supply a connection method (e.g., USB HID, UDP).
     - If a simulator is selected, the user must select which simulator implementation (e.g., Bullet, PhysX). The IDE must ask the kernel what simulator plugins are available.
+- One or more connection methods may be specified. If more than once connection method is specified, then each method is tried in order. If one method fails, the next method in the list is tried. If all methods fail, an error is thrown.
 
 ### Run Scripts
 
 - Users can create a run configuration that runs a script (this is just the default Groovy run configuration).
-  - If the user runs in the debug configuration, the IDE should automatically configure remote debugging to connect to the kernel daemon.
+  - If the user runs in the debug configuration, the IDE should automatically configure remote debugging to connect to the kernel daemon. This must be supported regardless of whether the kernel is running on the same machine as the client or is running on a different machine that is reachable via TCP.
```

Comment:
Question, because I don't know too much about this: in the remote case, how does it know what port to connect to? Is that just a standard set by command-line by our launcher? Do we need a gRPC API for that (on the kernel or the launcher)?

Reply:
We should consider the TCP port as dynamic, so yes, I suppose we would.